domain-model-assistant/Assets/Components/Scripts/Diagram.cs

```diff
@@ -11,419 +11,549 @@
-  void Awake()
-  {
-    if (Application.platform == RuntimePlatform.WebGLPlayer)
+    // Awake is called once to initialize this game object
+    void Awake()
     {
-      _isWebGl = true;
+        if (Application.platform == RuntimePlatform.WebGLPlayer)
+        {
+            _isWebGl = true;
+        }
     }
-  }
-
-   // Start is called before the first frame update
-  void Start()
-  {
-    CanvasScaler = this.gameObject.GetComponent<CanvasScaler>();
-    targetOrtho = CanvasScaler.scaleFactor;
-    this.raycaster = GetComponent<GraphicRaycaster>();
-    GetRequest(GetCdmEndpoint);
-  }
-
-  // Update is called once per frame
-  void Update()
-  {
-    if ((_currentMode == CanvasMode.Default && InputExtender.MouseExtender.IsDoubleClick()) ||
-        (_currentMode == CanvasMode.AddingClass && InputExtender.MouseExtender.IsSingleClick()))
+
+    // Start is called before the first frame update
+    void Start()
     {
-      AddClass("Class" + compartmentedRectangles.Count, Input.mousePosition);
-      ActivateDefaultMode();
+        CanvasScaler = this.gameObject.GetComponent<CanvasScaler>();
+        targetOrtho = CanvasScaler.scaleFactor;
+        this.raycaster = GetComponent<GraphicRaycaster>();
+        GetRequest(GetCdmEndpoint);
+
+
+        /* FOR UNITY FRONTEND DEVELOPMENT ONLY ie NO-BACKEND-SERVER*/
+        // LoadData();
+        //---------------------------------------//
     }
 
-    Zoom();
+    // Update is called once per frame
+    void Update()
+    {
+        if ((_currentMode == CanvasMode.Default && InputExtender.MouseExtender.IsDoubleClick()) ||
+            (_currentMode == CanvasMode.AddingClass && InputExtender.MouseExtender.IsSingleClick()))
+        {
+            AddClass("Class" + (compartmentedRectangles.Count+1), Input.mousePosition);
+            ActivateDefaultMode();
+        }
+
+        Zoom();
 
-    if (UseWebcore && _updateNeeded)
+        if (UseWebcore && _updateNeeded)
+        {
+            if (_getRequestAsyncOp != null && _getRequestAsyncOp.isDone)
+            {
+                var req = _getRequestAsyncOp.webRequest;
+                if (req.downloadHandler != null && !ReferenceEquals(req.downloadHandler, null))
+                {
+                    var newResult = req.downloadHandler.text;
+                    if (newResult != _getResult)
+                    {
+                        LoadJson(newResult);
+                        _getResult = newResult;
+                    }
+                }
+                _updateNeeded = false;
+                req.Dispose();
+            }
+            if (_postRequestAsyncOp != null && _postRequestAsyncOp.isDone)
+            {
+                _postRequestAsyncOp.webRequest.Dispose();
+            }
+            if (_deleteRequestAsyncOp != null && _deleteRequestAsyncOp.isDone)
+            {
+                _deleteRequestAsyncOp.webRequest.Dispose();
+            }
+            if (_putRequestAsyncOp != null && _putRequestAsyncOp.isDone)
```

Comment:
I think there is a bug here, even before this pull request. See #66 for details.